samples/command/consumer/src/main.rs

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+mod consumer_impl;
+
+use dtdl_parser::dtmi::{create_dtmi, Dtmi};
+use dtdl_parser::model_parser::ModelParser;
+use env_logger::{Builder, Target};
+use log::{info, LevelFilter};
+use proto::consumer::consumer_server::ConsumerServer;
+use proto::digitaltwin::digital_twin_client::DigitalTwinClient;
+use proto::digitaltwin::FindByIdRequest;
+use proto::provider::provider_client::ProviderClient;
+use proto::provider::InvokeRequest;
+use std::net::SocketAddr;
+use std::thread;
+use std::time;
+use tonic::transport::Server;
+use uuid::Uuid;
+
+/// The id for send notification command.
+const SEND_NOTIFICATION_COMMAND_ID: &str = "dtmi:org:eclipse:sdv:command:HVAC:send_notification;1";
+
+/// The id for the URI property.
+const URI_PROPERTY_ID: &str = "dtmi:sdv:property:uri;1";
+
+/// Start the ambient air temperature data stream.
+///
+/// # Arguments
+/// `provider_uri` - The provider_uri.
+/// `consumer_uri` - The consumer_uri.
+fn start_send_notification_repeater(provider_uri: String, consumer_uri: String) {
+    info!("Starting the Consumer's send notification repeater.");
+    tokio::spawn(async move {
+        loop {
+            info!("Invoking the send_notification command on endpoint {}", &provider_uri);
+
+            let client_result = ProviderClient::connect(provider_uri.clone()).await;
+            if client_result.is_err() {
+                continue;
+            }
+            let mut client = client_result.unwrap();
+
+            let response_id = Uuid::new_v4().to_string();
+
+            let payload: String = String::from("The send_notification request.");
+
+            let request = tonic::Request::new(InvokeRequest {
+                entity_id: String::from(SEND_NOTIFICATION_COMMAND_ID),
+                consumer_uri: consumer_uri.clone(),
+                response_id,
+                payload,
+            });
+
+            let _response = client.invoke(request).await;
+
+            thread::sleep(time::Duration::from_millis(1000));
```

Comment:
why not use async, for example, https://docs.rs/tokio/latest/tokio/time/fn.sleep.html? This code already uses async/await partially.

Reply:
Thanks!  Fixed.